File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    
    # 1. 각 작업의 완료까지 필요한 날짜 계산
    days = []
    for p, s in zip(progresses, speeds):
        days.append((100 - p + s - 1) // s)
    
    # 2. 배포 묶기
    current_day = days[0]
    count = 1
    
    for i in range(1, len(days)):
        if days[i] <= current_day:
            # 이전 기능과 함께 배포
            count += 1
        else:
            # 새로운 배포 시작
            answer.append(count)
            current_day = days[i]
            count = 1
    
    # 마지막 배포 추가
    answer.append(count)
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> result = new ArrayList<>();
        
        // 1. 각 작업 완료까지 걸리는 날짜 계산
        int[] days = new int[progresses.length];
        for (int i = 0; i < progresses.length; i++) {
            days[i] = (100 - progresses[i] + speeds[i] - 1) / speeds[i];
        }
        
        // 2. 배포 묶기
        int currentDay = days[0];
        int count = 1;
        
        for (int i = 1; i < days.length; i++) {
            if (days[i] <= currentDay) {
                count++;
            } else {
                result.add(count);
                currentDay = days[i];
                count = 1;
            }
        }
        
        // 마지막 배포
        result.add(count);
        
        // List → int[]
        int[] answer = new int[result.size()];
        for (int i = 0; i < result.size(); i++) {
            answer[i] = result.get(i);
        }
        
        return answer;
    }
}
